src/background/Wallet/Wallet.ts

```
@@ -131,6 +131,36 @@ async function prepareNonce<T extends { nonce?: BigNumberish; from?: string }>(
   }
 }
 
+function prepareTransactionType<
+  T extends {
+    type?: number;
+    maxFeePerGas?: BigNumberish;
+    maxPriorityFeePerGas?: BigNumberish;
+    gasPrice?: BigNumberish;
+  }
+>(transaction: T) {
+  if (transaction.type != null) {
+    return transaction;
+  }
+  if (
+    transaction.maxFeePerGas != null &&
+    transaction.maxPriorityFeePerGas != null
+  ) {
+    return {
+      ...transaction,
+      type: 2,
+    };
+  }
+  if (transaction.gasPrice != null) {
+    return {
+      ...transaction,
+      type: 0,
+    };
+  }
+  // we weren't able to populate transaction's type
```

Should we throw a descriptive error in this case? Just thinking

---

Ah, nvm, it doesn't make sense to assume it would fail